### ISSUE TITLE
BUG: No matching `ray` package when pip installing dev dependencies

### ISSUE THREAD

**Issue by NickCrews** [modin-project/modin#4579]
### System information
- **OS Platform and Distribution **: MacOS 12.4
- **Modin version**: `main`, [4ec7f63](https://github.com/modin-project/modin/commit/4ec7f6347903f9133c65ebc5b6e0e15553b98577)
- **Python version**: 3.10.4
- **Code we can use to reproduce**: per https://modin.readthedocs.io/en/stable/development/contributing.html#development-dependencies, running

```bash
python -m venv .venv
source .venv/bin/activate
pip install -r requirements-dev.txt
```

### Describe the problem
pip can't find a matching version of Ray from the requirements in dev-dependencies. Probably caused by https://github.com/modin-project/modin/issues/4566

I can't install the dev environment, and therefore run tests or otherwise work on https://github.com/modin-project/modin/pull/4571.

### Source code / logs
```
$ pip install -r requirements-dev.txt
Collecting git+https://github.com/airspeed-velocity/asv.git@ef016e233cb9a0b19d517135104f49e0a3c380e9 (from -r requirements-dev.txt (line 33))
  Cloning https://github.com/airspeed-velocity/asv.git (to revision ef016e233cb9a0b19d517135104f49e0a3c380e9) to /private/var/folders/nz/wfwcn6md1796t48tx9dhf0v40000gn/T/pip-req-build-m18f3xa6
  Running command git clone --filter=blob:none --quiet https://github.com/airspeed-velocity/asv.git /private/var/folders/nz/wfwcn6md1796t48tx9dhf0v40000gn/T/pip-req-build-m18f3xa6
  Running command git rev-parse -q --verify 'sha^ef016e233cb9a0b19d517135104f49e0a3c380e9'
  Running command git fetch -q https://github.com/airspeed-velocity/asv.git ef016e233cb9a0b19d517135104f49e0a3c380e9
  Running command git checkout -q ef016e233cb9a0b19d517135104f49e0a3c380e9
  Resolved https://github.com/airspeed-velocity/asv.git to commit ef016e233cb9a0b19d517135104f49e0a3c380e9
  Running command git submodule update --init --recursive -q
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting pandas==1.4.2
  Using cached pandas-1.4.2-cp310-cp310-macosx_10_9_x86_64.whl (11.1 MB)
Collecting numpy>=1.18.5
  Using cached numpy-1.22.4-cp310-cp310-macosx_10_15_x86_64.whl (17.7 MB)
Collecting pyarrow>=4.0.1
  Using cached pyarrow-8.0.0-cp310-cp310-macosx_10_13_x86_64.whl (22.4 MB)
Collecting dask[complete]<2022.2.0,>=2.22.0
  Using cached dask-2022.1.1-py3-none-any.whl (1.1 MB)
Collecting distributed<2022.2.0,>=2.22.0
  Using cached distributed-2022.1.1-py3-none-any.whl (830 kB)
ERROR: Could not find a version that satisfies the requirement ray[default]<1.13.0,>=1.4.0 (from versions: 1.13.0)
ERROR: No matching distribution found for ray[default]<1.13.0,>=1.4.0
```

**Comment by mvashishtha** [comment]
@NickCrews it looks like the only version of ray that is available on python 3.10 is 0.13.0, but ray 0.13.0 is not compatible with  modin yet due to #4564 . Is it possible for you to do your work in an environment that uses python 3.9.12, where ray 1.12.1, the latest version before 0.13.0, is available?

**Comment by NickCrews** [comment]
@mvashishtha Thank you, that was a very easy fix. That works fine for me now!
